recipes/zserio/all/conanfile.py

```diff
@@ -0,0 +1,79 @@
+import os
+import shutil
+
+from conan import ConanFile
+from conan.tools.files import copy, get, collect_libs
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.scm import Version
+from conan.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=2.0.0"
+
+class ZserioConanFile(ConanFile):
+    name = "zserio"
+    description = "Zserio C++ Runtime Library"
+    license = "BSD-3 Clause"
+    url = "https://github.com/conan-io/conan-center-index/"
+    homepage = "https://zserio.org"
+    topics = ("zserio", "C++", "serialization")
+    package_type = "static-library"
+    settings = "os", "arch", "compiler", "build_type"
+
+    @property
+    def _min_cppstd(self):
+        return 11
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if Version(self.version) < "2.13.0":
+            raise ConanInvalidConfiguration("Minimum available version is 2.13.0")
+
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version]["runtime"],
+            pattern="runtime_libs/cpp/*", strip_root=True)
+        shutil.move("cpp/zserio", ".")
+        shutil.move("cpp/CMakeLists.txt", ".")
+        shutil.rmtree("cpp")
+
+    def export_sources(self):
+        copy(self, "zserio_compiler.cmake", self.recipe_folder, self.export_sources_folder)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+        get(self, **self.conan_data["sources"][self.version]["compiler"], pattern="zserio.jar")
+
+    @property
+    def _cmake_module_path(self):
+        return os.path.join("lib", "cmake", "zserio")
+
+    def package(self):
+        include_dir = os.path.join(self.package_folder, "include")
+        lib_dir = os.path.join(self.package_folder, "lib")
+        copy(self, "*.h", self.source_folder, include_dir)
+        copy(self, "*.lib", self.build_folder, lib_dir)
+        copy(self, "*.a", self.build_folder, lib_dir)
+
+        copy(self, "zserio.jar", self.build_folder, os.path.join(self.package_folder, "bin"))
+        copy(self, "zserio_compiler.cmake", self.export_sources_folder,
+             os.path.join(self.package_folder, self._cmake_module_path))
+
+    def package_info(self):
+        self.cpp_info.components["ZserioCppRuntime"].libs = collect_libs(self)
```

Comment:
```suggestion
        self.cpp_info.components["ZserioCppRuntime"].libs = collect_libs(self)
```
Please, do not use `collect_libs` it hides what should be copied and does not respect linkage order in case of multiple libraries.

Reply:
I found it easier than guessing the static library suffix which can be either `.a` or `.lib` and I didn't want to make a complicated platform checks as I found in some other recipes 
Windows MSVC -> lib
Windows MinGW -> a
Linux -> a
...
Is there some clean solution to this? Unfortunately the current zserio `CMakeLists.txt` doesn't have the install step defined.

Reply:
btw. `collect_libs` seems to be used also in other recipes (found 110x)

Reply:
@Mi-La Please, take a look in the cmake project template as example: https://github.com/conan-io/conan-center-index/blob/7fdbb2410f2b8e682bfc26aee7b6a0a53ad93bed/docs/package_templates/cmake_package/all/conanfile.py You do not need to specify the library extension neither its prefix. For instance: if your library name is `foobar`, it may produce `libfoobar.so` on Linux and `foobar.lib` on Windows, resulting in `self.cpp_info.libs = ["foobar"]` Conan is smart enough to understand it.

Reply:
> btw. collect_libs seems to be used also in other recipes (found 110x)

Yes, it's part of the legacy, unfortunately. But well discouraged nowadays. We had problems using `collect_libs` in the past, related to linker order. Plus, is harder to debug when we don't know which library should be installed.

Reply:
Oh, this is exactly the feature I needed :-). I was looking to many other recipes and I've always seen that they are guessing the "lib" prefix. Also in the [doc](https://docs.conan.io/2/reference/conanfile/methods/package_info.html#components) there is "libcrypto" used instead of just "crypto".
However if conan can deal with just the lib-name, it's great!